src/main/flight/pid_init.c

```diff
@@ -66,6 +68,61 @@ static void pidSetTargetLooptime(uint32_t pidLooptime)
 #endif
 }
 
+
+#ifdef USE_WING
+void tpaSpeedEstBasicInit(const pidProfile_t *pidProfile)
+{
+    const float gravityFactor =  pidProfile->tpa_speed_est_basic_gravity / 100.0f;
+    const float delaySec = pidProfile->tpa_speed_est_basic_delay / 1000.0f;
+
+    pidRuntime.tpaSpeedEst.twr = 1.0f / (gravityFactor * gravityFactor);
+    pidRuntime.tpaSpeedEst.massDragRatio = (2.0f/logf(3.0f)) * (2.0f/logf(3.0f)) * pidRuntime.tpaSpeedEst.twr * G_ACCELERATION * delaySec * delaySec;
+    pidRuntime.tpaSpeedEst.maxSpeed = sqrtf(pidRuntime.tpaSpeedEst.massDragRatio * pidRuntime.tpaSpeedEst.twr * G_ACCELERATION + G_ACCELERATION);
+}
+
+
+void tpaSpeedEstAdvancedInit(const pidProfile_t *pidProfile)
+{
+    pidRuntime.tpaSpeedEst.twr = (float)pidProfile->tpa_speed_est_adv_thrust/(float)pidProfile->tpa_speed_est_adv_mass;
```

Comment:
Thrust to weight ratio misses the gravitational acceleration. (weight = mass * gravity).
```suggestion
    const float weight = pidProfile->tpa_speed_est_adv_mass * G_ACCELERATION;
    pidRuntime.tpaSpeedEst.twr = pidProfile->tpa_speed_est_adv_thrust / weight;
```

Reply:
> Thrust to weight ratio misses the gravitational acceleration. (weight = mass * gravity).

in CLI the `tpa_speed_est_adv_thrust` and `tpa_speed_est_adv_mass` both just in grams. So if a plane is 1000g weight but 2000g max thrust, then twr is 2.0.

Reply:
Measuring prop thrust as masses would be odd. I think you are referring to kilogramm-force instead. That is still a force in Newton but normalized to the weight of 1kg of mass. So 1kgf is equal to 9.8N. In that case I'm satisfied.

Reply:
> Measuring prop thrust as masses would be odd. I think you are referring to kilogramm-force instead. That is still a force in Newton but normalized to the weight of 1kg of mass. So 1kgf is equal to 9.8N. In that case I'm satisfied.

it's just common when bragging about how powerful a motor is to say "it provides 2kg thrust" :) Nobody says "19.6 newtons" in the hobby, and all the thrust-amp consumption tables for motors are in grams.

In the pidloop to calculate thrust **acceleration** i'm doing doing:
`const float thrust = (throttle * throttle - throttle * tpa->speed / tpa->propMaxSpeed) * tpa->twr * G_ACCELERATION;`